File: cpp/sorting/standard/selection_sort.cpp
#include <vector>

using namespace std;

vector<int> selection_sort(const vector<int>& arr) {
    vector<int> a = arr;
    int n = (int)a.size();
    for (int i = 0; i < n; i++) {
        int min_idx = i;
        for (int j = i + 1; j < n; j++) {
            if (a[j] < a[min_idx]) min_idx = j;
        }
        if (min_idx != i) swap(a[i], a[min_idx]);
    }
    return a;
}

File: java/sorting/standard/SelectionSort.java
import java.util.Arrays;

public class SelectionSort {

    public static int[] selectionSort(int[] arr) {
        int[] a = Arrays.copyOf(arr, arr.length);
        int n = a.length;

        for (int i = 0; i < n; i++) {
            int minIdx = i;

            for (int j = i + 1; j < n; j++) {
                if (a[j] < a[minIdx]) {
                    minIdx = j;
                }
            }

            if (minIdx != i) {
                int temp = a[i];
                a[i] = a[minIdx];
                a[minIdx] = temp;
            }
        }

        return a;
    }
}
